omod/src/test/webapp/resources/scripts/navigatorHandlers.js

```diff
@@ -433,14 +434,72 @@ describe("Tests for simple form navigation handlers", function() {
+            firstQuestion.isSelected = true;
+            spyOn(firstQuestion, 'isValid').andReturn(true);
+            spyOn(firstQuestion, 'onExit').andReturn(true);
+            spyOn(secondQuestion, 'isDisabled').andReturn(false);
+            spyOn(firstQuestion, 'toggleSelection');
+            spyOn(secondQuestion, 'toggleSelection');
+
+            questionsHandler = new QuestionsHandler([firstQuestion, secondQuestion], prevButton);
+            var wasHandled = questionsHandler.nextQuestion();
+            expect(prevButton.hide).not.toHaveBeenCalled();
+            expect(prevButton.show).toHaveBeenCalled();
+          });
+
+
+          it("should hide previous button when jumping from second question to first question", function() {
+            secondQuestion.isSelected = true;
+            spyOn(secondQuestion, 'isValid').andReturn(true);
+            spyOn(secondQuestion, 'onExit').andReturn(true);
+            spyOn(firstQuestion, 'isDisabled').andReturn(false);
+            spyOn(firstQuestion, 'toggleSelection');
+            spyOn(secondQuestion, 'toggleSelection');
+
+            questionsHandler = new QuestionsHandler([firstQuestion, secondQuestion], prevButton);
+            var wasHandled = questionsHandler.prevQuestion();
+            expect(prevButton.hide).toHaveBeenCalled();
+            expect(prevButton.show).not.toHaveBeenCalled();
+          });
+
+          it("should not hide or show previous button when jumping from second question to third question", function() {
+            secondQuestion.isSelected = true;
+            spyOn(secondQuestion, 'isValid').andReturn(true);
+            spyOn(secondQuestion, 'onExit').andReturn(true);
+            spyOn(thirdQuestion, 'isDisabled').andReturn(false);
+            spyOn(thirdQuestion, 'toggleSelection');
+            spyOn(secondQuestion, 'toggleSelection');
+
+            questionsHandler = new QuestionsHandler([firstQuestion, secondQuestion], prevButton);
+            var wasHandled = questionsHandler.nextQuestion();
+            expect(prevButton.hide).not.toHaveBeenCalled();
+            expect(prevButton.show).not.toHaveBeenCalled();
+          });
+
+          it("should not hide or show previous button when jumping from third question to second question", function() {
+            thirdQuestion.isSelected = true;
+            spyOn(thirdQuestion, 'isValid').andReturn(true);
+            spyOn(thirdQuestion, 'onExit').andReturn(true);
+            spyOn(secondQuestion, 'isDisabled').andReturn(false);
+            spyOn(secondQuestion, 'toggleSelection');
+            spyOn(thirdQuestion, 'toggleSelection');
+
+            questionsHandler = new QuestionsHandler([firstQuestion, secondQuestion], prevButton);
+            var wasHandled = questionsHandler.prevQuestion();
+            expect(prevButton.hide).not.toHaveBeenCalled();
+            expect(prevButton.show).not.toHaveBeenCalled();
+          });
+
```

Comment:
Maybe, to mark it as pending, something like
```suggestion

          xit("should go to beginning and hide the previous button after Cancelling from Confirmation");
```
?

Reply:
This shouldn't be pending... have I flagged it as such somewhere?

Reply:
Oh, there was a "wasHandled" assignment that got left in when I copied and pasted from another test, don't know if that was what was confusing? I removed it.